### DISCUSSION TITLE
Can a single entt::observer track if a component is either added or updated?

### DISCUSSION THREAD

**Discussion by Net5F** [skypjack/entt#1068] [Q&A] [answered]
Hello! This answer might be obvious but I'm having trouble figuring it out.

Can a single `entt::observer` be used to track if a given component type is either added to an entity _or_ updated? Or does this require 2 observers?

For context, I need to end up with a list of unique entities that had a given component type added or updated. If 2 observers are indeed required, is the best approach to sum their lists, sort, and remove duplicates?

**Comment by skypjack** [comment, marked as answer]
Uhm, out of my mind, it should be `entt::collector.group<T>().update<T>()`. Have you tried it already?

**Reply by Net5F** [reply]
Looks like that works, thank you!

**Reply by Net5F** [reply]
On MSVC everything works great, but GCC needed some help with parsing:
```
entt::collector.group<T>().template update<T>()
```
Unfortunately it also has a warning:
```
/.../Libraries/entt/src/entt/entity/registry.hpp:983:49: warning: parameter ‘entt’ set but not used [-Wunused-but-set-parameter]
  983 |     [[nodiscard]] bool any_of(const entity_type entt) const {
      |                               ~~~~~~~~~~~~~~~~~~^~~~
```
It would be nice if there was a way to get rid of the warning, but it isn't too big of a deal.

**Reply by skypjack** [reply]
Oh, good catch. The class invokes `any_of` and the latter also accepts an empty parameter pack.
In this case, `entt` isn't used actually. A `[[maybe_unused]]` is enough but it's not there in your version.
Adding it upstream though. Completely missed so far. Sorry.

**Reply by Net5F** [reply]
No worries! I appreciate you adding the fix, and also thanks for building all of this 😄 I'm wrapping up this work, then I'll be excited to share some of the cool stuff this library has allowed me to do.
